pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
```

Comment:
I don't *think* they need to be different.  I guess the question is whether
people want to be explicit with secrets or not.  Since the schema looks
like it handles both types, I think the type should also be generic.  We
might in the future add a new secret schema type (tls-server) if we had
to.  Another thing to keep in mind - if we get into serving certs, every
programming language has its own cert container story (Java -> Go ->
OpenSSL) so we may also need to qualify in the future the type.

On Thu, Feb 4, 2016 at 12:30 PM, Prashanth B <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/20142#discussion_r51906838>
> :
>
> > @@ -2143,6 +2143,20 @@ const (
> >
> >  	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
> >  	SSHAuthPrivateKey = "ssh-privatekey"
> > +
> > +	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
> > +	//
> > +	// Required fields:
> > +	// - Secret.Data["key"] - TLS private key.
> > +	//   Secret.Data["cert"] - TLS certificate.
> > +	//   Secret.Data["chain"] - optional certificate chain.
> > +	// TODO: Consider supporting different formats, specifying CA/destinationCA.
> > +	SecretTypeTLS SecretType = "kubernetes.io/tls"
>
> That was part of my motivation for including "Chain" and being all
> hand-wavy in the comments about what chain it was. By client cert I'm
> assuming you mean the cert a server would use to re-encrypt (or the cert a
> client would use for client auth), does they need to be different?
>
> Or were you talking about something from the client that might include a
> csr etc?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20142/files#r51906838>.
>